src/se/terhol/pisemka31/VendorImpl.java

```diff
@@ -0,0 +1,58 @@
+package se.terhol.pisemka31;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+/**
+ * @author Tereza Holm
+ */
+public class VendorImpl implements Vendor {
+    SortedMap<IceCream, Integer> iceCreams = new TreeMap<>();
+
+    @Override
+    public void sell(IceCream iceCream, int required) throws IceCreamException {
```

Comment:
Why does it throw `IceCreamException ` instead of `OutOfStockException`? But it looks like a mistake in interface

Reply:
I am not sure when it is supposed to throw IceCreamException, interface does not specify that.

Reply:
Well this methods throws just `OutOfStockException` and any `IceCreamException`, that's why I am wondering... But as I said it is ok, it is probably a mistake in the task description